Modules/ITS/src/ITSThresholdCalibrationTask.cxx

```diff
@@ -0,0 +1,456 @@
+    hCalibrationChipAverage[iScan][iBarrel]->Divide(hCalibrationChipSum[iScan][iBarrel], hCalibrationChipCounts[iScan][iBarrel]);
+    hCalibrationRMSChipAverage[iScan][iBarrel]->Divide(hCalibrationRMSChipSum[iScan][iBarrel], hCalibrationChipCounts[iScan][iBarrel]);
+
+    if (iScan == 2) {
+      hCalibrationThrNoiseChipAverage[iBarrel]->Divide(hCalibrationThrNoiseChipSum[iBarrel], hCalibrationChipCounts[iScan][iBarrel]);
+      hCalibrationThrNoiseRMSChipAverage[iBarrel]->Divide(hCalibrationThrNoiseRMSChipSum[iBarrel], hCalibrationChipCounts[iScan][iBarrel]);
+    }
+  }
+
+  for (int iLayer = 0; iLayer < 7; iLayer++) { // TH1 plots per layer
+
+    Int_t iBarrel = getBarrel(iLayer);
+
+    for (int iStave = 1; iStave <= NStaves[iLayer]; iStave++)
+      for (int iChip = 1; iChip <= nChipsPerStave[iLayer]; iChip++) {
+        hCalibrationLayer[iLayer][iScan]->Fill(hCalibrationChipAverage[iScan][iBarrel]->GetBinContent(iChip, StaveBoundary[iLayer] + iStave));
+        hCalibrationRMSLayer[iLayer][iScan]->Fill(hCalibrationRMSChipAverage[iScan][iBarrel]->GetBinContent(iChip, StaveBoundary[iLayer] + iStave));
+
+        if (iScan == 2) {
+          hCalibrationThrNoiseLayer[iLayer]->Fill(hCalibrationThrNoiseChipAverage[iBarrel]->GetBinContent(iChip, StaveBoundary[iLayer] + iStave));
+          hCalibrationThrNoiseRMSLayer[iLayer]->Fill(hCalibrationThrNoiseChipAverage[iBarrel]->GetBinContent(iChip, StaveBoundary[iLayer] + iStave));
+        }
+      }
+  }
+}
+
+ITSThresholdCalibrationTask::CalibrationResStruct ITSThresholdCalibrationTask::CalibrationParser(string input)
+{
+  CalibrationResStruct result;
+  auto StaveINFO = split(input, ",");
+
+  for (string info : StaveINFO) {
+    if (info.size() == 0)
+      continue;
+
+    if (info.substr(0, 1) == "L") {
+      result.Layer = std::stod(info.substr(1, 1));
+      result.Stave = std::stod(info.substr(3, 2));
+    } else {
+
+      std::string name = split(info, ":")[0];
+      string data = split(info, ":")[1];
+      if (name == "Hs_pos") {
+        result.Hs = std::stod(data);
+      } else if (name == "Hic_Pos") {
+        result.HIC = std::stod(data);
+      } else if (name == "ChipID") {
+        result.ChipID = std::stod(data);
+      } else if (name == "VCASN") {
+        result.VCASN = std::stof(data);
+      } else if (name == "Rms") {
+        result.RMS = std::stof(data);
+      } else if (name == "Status") {
+        result.status = std::stod(data);
+      } else if (name == "Noise") {
+        result.Noise = std::stof(data);
+      } else if (name == "NoiseRMS") {
+        result.NoiseRMS = std::stof(data);
+      } else if (name == "ITHR") {
+        result.ITHR = std::stof(data);
+      } else if (name == "THR") {
+        result.THR = std::stof(data);
+      }
+    }
+  }
+  return result;
+}
+
+void ITSThresholdCalibrationTask::endOfCycle()
+{
+
+  std::ifstream runNumberFile(mRunNumberPath.c_str());
+  if (runNumberFile) {
```

Comment:
I see that in some tasks we do not fill metadata to the TH objects at all. 

For example, it is ignored in FhrTask, FeeTask, while we have metadata in the Cluster and Track tasks. Other Modules do not have an agreement as well.

Do we need to fill metadata or it can be skipped?

Reply:
You can remove this because even if in some tasks is still there, we don't use it anymore (i.e. in the json we are not setting anymore the runNumberPath variable). Now the run number is automatically retrieved by ECS logic.